Implement MLP for Regression

SebastianOndrus opened the pull request:
This pull request addresses issue #15: "Implement MLP for Regression".

**Implemented Features**:
  - **MLP Regression**: Added a implementation of MLP regression to the  `multilayer_perceptron.py` file.
  - **Tests**: Integrated test into `test_mlp.py` file to ensure the correct functionality of the MLP regression.
  - **Examples**:  Updated the `mlp.py` file (same file as the MLP classifier example) to include an example demonstrating the usage of the MLP regression).

Please review the changes and provide any feedback or suggestions.

rafaelgreca commented on scratchml/models/multilayer_perceptron.py:
```diff
@@ -333,6 +350,12 @@ def fit(self, X: np.ndarray, y: np.ndarray) -> None:
                         loss = cross_entropy(batch_y, y_hat, derivative=True)
+                elif isinstance(self, MLPRegressor):
+                    if self.loss_function == "mse":
+                        loss = 2 * (y_hat - batch_y) / len(batch_y)  # Derivative of MSE w.r.t predictions
```
Please, update that line to use the "mean_squared_function" defined in line 4 on the metrics.py file. For example:

```python
loss = 2 * mean_squared_error(batch_y, y_hat, derivative=True) / len(batch_y)
```

SebastianOndrus replied:
Done!

rafaelgreca commented on scratchml/models/multilayer_perceptron.py:
```diff
@@ -333,6 +350,12 @@ def fit(self, X: np.ndarray, y: np.ndarray) -> None:
+                    if self.loss_function == "mse":
+                        loss = 2 * (y_hat - batch_y) / len(batch_y)  # Derivative of MSE w.r.t predictions
+
+                if loss is None:
```
Did you try to set loss equal to None? Did the code run? If so, it's better to set that validation in the init function, such as:

```python
# Validate the loss function
        if not self.loss_function:
            if self.loss_function not in self._valid_loss_functions:
                raise ValueError(
                    f"The value for 'loss_function' must be one of {self._valid_loss_functions}."
                )
        else:
            raise ValueError("Loss function computation failed. Check configuration.")
```

If the code did not run and raised a ValueError, you can just delete both lines.

SebastianOndrus replied:
Both lines have been removed.

rafaelgreca reviewed (changes requested):
Sorry, I forgot to add to the GitHub workflow, but can you please use the [Black](https://github.com/psf/black) code formatter (you only need to install it and then run ```black . --safe``` on the root folder) ? Besides that, I tested your code, and it's running smoothly! Thanks so much for your contribution! Once you review my suggestions and commit again, I'll merge the code.

Oh, before I forget, can you please check/complete both "MLP" and "MLP Regressor" options on the "Roadmap" section in the "README" file? Also, feel free to create a paragraph about yourself (just like I did) in the "AUTHORS" file inside the "docs" folder (you can put it on the "OTHER CONTRIBUTORS" section).

Again, thanks so much for your contribution to the project! Whenever you want to, feel free to contribute to other issues! :D

SebastianOndrus commented on tests/models/test_mlp.py:
```diff
@@ -76,7 +76,7 @@ def test_2(self):
         mlp = MLPClassifier(loss_function="bce", hidden_layer_sizes=(32, 64, 128))
         skmlp = SkMLP(
             hidden_layer_sizes=(32, 64, 128),
-            solver="sgd",
```
Done!

SebastianOndrus commented:
Hi! @rafaelgreca 
I've addressed all the requested changes:

- Changed code based on feedback.
- Ran Black for code formatting.
- Updated the README 'Roadmap' section for MLP and MLP Regressor.
- Added myself to the AUTHORS file.

Thanks again for the helpful feedback. Let me know if there's anything else you'd like me to adjust!"
